src/routes/notifications/v1/entities/__tests__/create-registration-v2.dto.builder.ts

```diff
@@ -0,0 +1,73 @@
+import type { UpsertSubscriptionsDto } from '@/datasources/notifications/entities/upsert-subscriptions.dto.entity';
+import { AuthPayload } from '@/domain/auth/entities/auth-payload.entity';
+import { NotificationType } from '@/domain/notifications/v2/entities/notification.entity';
+import type { RegisterDeviceDto } from '@/routes/notifications/v1/entities/register-device.dto.entity';
+import type { UpsertSubscriptionsSafesDto } from '@/routes/notifications/v2/entities/upsert-subscriptions.dto.entity';
+import type { UUID } from 'crypto';
+import { recoverMessageAddress } from 'viem';
+
+export const createV2RegisterDtoBuilder = async (
+  args: RegisterDeviceDto,
+): Promise<
+  Array<{
+    upsertSubscriptionsDto: UpsertSubscriptionsDto;
+    authPayload: AuthPayload;
+  }>
+> => {
+  const safeV2Array: Array<{
+    authPayload: AuthPayload;
+    upsertSubscriptionsDto: UpsertSubscriptionsDto & {
+      signature: `0x${string}`;
+    };
+  }> = [];
+
+  const safesV1Registrations = args.safeRegistrations;
+
+  for (const safeV1Registration of safesV1Registrations) {
+    if (safeV1Registration.safes.length) {
+      const safeV2: {
+        authPayload: AuthPayload;
+        upsertSubscriptionsDto: UpsertSubscriptionsDto & {
+          safes: Array<UpsertSubscriptionsSafesDto>;
+          signature: `0x${string}`;
+        };
+      } = {
+        upsertSubscriptionsDto: {
+          cloudMessagingToken: args.cloudMessagingToken,
+          deviceType: args.deviceType,
+          deviceUuid: (args.uuid as UUID) || undefined,
```

Comment:
Why do we need to add `undefined`?

Reply:
Then `deviceUUid:` will be undefined. I don't understand how adding `| undefined` adds type safety. Could you provide an example to clarify?

Reply:
It is the inherent type that is passed in `args` - RegisterDeviceDto['uuid'], or `string | undefined`. You are otherwise casting to something it is not, an always defined `UUID`:

```ts
const entity = new RegisterDeviceDto(
  undefined, // uuid
  'cloudMessagingToken',
  'buildNumber',
  'bundle',
  DeviceType.ANDROID,
  'version',
  undefined,
  [],
);
```

Reply:
I know what you mean but then `(args.uuid as UUID) || undefined` will be undefined if `args.uuid = ''`. Anyway, I'll add the extra undefined.

Reply:
Since the PR is approved, I'm gonna resolve this thread.